### PR TITLE
Adding IoC class

### PR DESCRIPTION
**Pull request by disraelyp**

### CODE REVIEW

File: src/main/java/com/disraely/exceptions/ioc/KeyNullPointerException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.disraely.exceptions.ioc;
+
+public class KeyNullPointerException extends RuntimeException {
+  public KeyNullPointerException() {
+    super("The Key can not be null");
+  }
+}
```
Comment (reynoldmorel) [inline src/main/java/com/disraely/exceptions/ioc/KeyNullPointerException.java]: Add enter

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>${mavencheckstyleplugin.version}</version>
+                <dependencies>
+                    <dependency>
+                        <groupId>com.puppycrawl.tools</groupId>
+                        <artifactId>checkstyle</artifactId>
+                        <version>10.3.4</version>
```
Comment (reynoldmorel) [inline pom.xml:35]: Add property for this one

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (reynoldmorel) [inline pom.xml:1]: In general is there a a maven task for formatting? I can't see it

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/main/java/com/disraely/exceptions/ioc/KeyNullPointerException.java (hunk 1 above)
Comment (disraelyp) [reply src/main/java/com/disraely/exceptions/ioc/KeyNullPointerException.java]: Resolved!

File: pom.xml (hunk 2 above)
Comment (disraelyp) [reply pom.xml:35]: Resolved!

**Pull request closed without merging**